crates/astria-composer/src/executor/bundle_factory/mod.rs

```diff
@@ -153,14 +184,34 @@ impl BundleFactory {
                 })
             }
             Err(SizedBundleError::NotEnoughSpace(seq_action)) => {
-                // if the bundle is full, flush it and start a new one
-                self.finished.push_back(self.curr_bundle.flush());
-                self.curr_bundle
-                    .push(seq_action)
-                    .expect("seq_action should not be larger than max bundle size, this is a bug");
-                Ok(())
+                if self.finished.len() >= self.finished_queue_capacity {
+                    Err(BundleFactoryError::FinishedQueueFull {
+                        curr_bundle_size: self.curr_bundle.curr_size,
+                        seq_action,
+                    })
+                } else {
+                    self.finished.push_back(self.curr_bundle.flush());
+                    // if the finished queue is full after the flush mark it as such
+                    if self.finished.len() == self.finished_queue_capacity {
+                        self.full = true;
+                    }
```

Comment:
I think we can just remove this logic here. If we define `full()` with `self.finished.len() > self.finished_queue_capacity`

Reply:
The `BundleFactory` can have a full `finished` queue but still have space in its `curr_bundle`.